main.tf

```diff
@@ -15,29 +20,20 @@ data "aws_route53_zone" "default" {
   private_zone = false
 }
 
-locals {
-  domain_validation_options = "${aws_acm_certificate.default.domain_validation_options[0]}"
-}
-
-resource "null_resource" "default" {
-  count = "${var.process_domain_validation_options == "true" && var.validation_method == "DNS" ? length(aws_acm_certificate.default.domain_validation_options) : 0}"
-
-  triggers = "${aws_acm_certificate.default.domain_validation_options[count.index]}"
-}
-
 resource "aws_acm_certificate_validation" "default" {
+  count           = "${var.process_domain_validation_options == "true" ? 1 : 0}"
   certificate_arn = "${aws_acm_certificate.default.arn}"
 
   validation_record_fqdns = [
-    "${distinct(compact(concat(aws_route53_record.default.fqdn, var.subject_alternative_names)))}",
+    "${distinct(compact(aws_route53_record.default.*.fqdn))}",
```

Comment:
This is nice.

Reply:
Its in use and is creating and destroying fine. :)